### DISCUSSION TITLE
packages之间的相互依赖，为什么显示的是固定版本号而非"workspace:*"

### DISCUSSION THREAD

**Discussion by GuanJdoJ** [ant-design/pro-components#8242] [Q&A]
在`pro-components`项目中，存在不同的`packages`之间相互依赖，比如`pro-form`的`package.json`中
```json
"dependencies": {
  "@ant-design/pro-field": "2.14.8",
}
```
此时的`@ant-design/pro-field`是一个固定的版本号，而不是
```json
"dependencies": {
  "@ant-design/pro-field": "workspace:*",
}
```

我知道`pnpm`在进行`workspace`包发布时，会动态替换`workspace:`依赖；
1.但是在进行开发时，如果`pro-field`和`pro-form`都进行了更改，那么`pro-form`是如何使用`pro-field`的最新代码呢，这里面有怎么样的机制呢？

2.我阅读了`/scripts/release.js`代码，了解了在进行发布时，`lerna`会帮助`bump version`和生成 `change log`，但是也没读到有关替换成固定版本的逻辑，难道这是`lerna version`自动帮我们做的吗，如果是这样，那么当我需要在`pro-from`中使用一个新的`package`: `pro-xxx`，该执行什么命令呢？我试了使用
```shell
pnpm --filter lib1 add lib2 --workspace
```
这样的脚本，生成的依赖关系也是
```json
"dependencies": {
  "@ant-design/pro-xxx": "workspace:^",
}
```

**Comment by GuanJdoJ** [comment]
不好意思打扰了，这块实在是没太明白，大佬有空时能不能帮忙解下惑🙏 @PeachScript

**Comment by warmhug** [comment]
@GuanJdoJ 知道原因了吗？有同样的疑惑

**Reply by GuanJdoJ** [reply]
我估计原因如下：
1 packages中的包都要发布到外部，这些包的依赖关系需要固定的版本号；
2.每次发布时也能明确地知道依赖的版本，这样能保证依赖关系清晰，避免因为依赖的最新变化产生问题；

lerna在这个过程中起版本管理的作用，当一个依赖A的版本变动，lerna会帮助我们自动将其他包中使用到依赖A的版本也转变成新的版本号，所以即使使用固定版本号，也不需要手动更改，lerna会帮助我们调整。

**Reply by warmhug** [reply]
主要是疑惑，为什么不写 ~或^ 号前缀
